File: assets/docs/configuration/transformations/custom-scripts/create-a-script-setpk-modify-example.lua
function main(input)
    -- input is a string, so we parse it
    local json = require("json")
    local jsonObj, _ = json.decode(input.Data)

    -- set the name field
    jsonObj.name = "Bruce Wayne"

    -- Pass it back to Stream Replicator via the Data field
    return { Data = jsonObj, ParititionKey = "myPk" }
end

File: assets/docs/configuration/transformations/custom-scripts/create-a-script-setpk-modify-example.js
function main(input) {
    // input is a string, so we parse it
    var jsonObj = JSON.parse(input.Data);
    
    // set the name field
    jsonObj.name = "Bruce Wayne"
    return {
        // Pass it back to Snowbridge via the Data field
        Data: jsonObj,
        PartitionKey: "myPk"
    };
}